presto-hive/src/main/java/com/facebook/presto/hive/InternalHiveSplit.java

```diff
@@ -120,7 +121,8 @@ public InternalHiveSplit(
 
     public String getPath()
     {
-        return new String(path, StandardCharsets.UTF_8);
+        String relativePathString = new String(relativeUri, StandardCharsets.UTF_8);
+        return new Path(partitionInfo.getPath().resolve(relativePathString)).toString();
```

Comment:
Why `new Path(...).toString()` is needed?

Reply:
The Path toString fuction is slightly different from the URI toString function, and I wanted to be sure the String we produce is the same.